### ISSUE TITLE
Chrome / IE Enumeration 

### ISSUE THREAD

**Issue by unl1k3ly** [byt3bl33d3r/CrackMapExec#112]
Hi, 
Would be possible to have as a module the --chrome-enum and ie-enum to download store/cached/cookies passwords from the browser ?

Hardest part would be to run as a chrome/ie user context, but I reckon it will be most likely users context from explorer.exe process, if migrated onto it.

Cheers

**Comment by byt3bl33d3r** [comment]
@vinicius777 It is very much possible and is on my feature list. I have some ideas on how to implement it, just need the time. Will try to push it out as a module asap.

**Comment by unl1k3ly** [comment]
Thanks mate!!

**Comment by AlessandroZ** [comment]
I don't know if you are interested, but I have developed a tool to do that and more: 
- https://github.com/AlessandroZ/LaZagne

It could be useful to retrieve passwords from lots of software. I will release a new version soon lighter and with new features (it will impersonate token to retrieve credentials from all users, etc.). 

Feel free to use your own code if you prefer.

**Comment by byt3bl33d3r** [comment]
Just added the enum_chrome module in 74f746592ace591a021c21087812fd5818d87b13, check it out! ;)

**Comment by unl1k3ly** [comment]
Awesome! Thanks mate

**Comment by byt3bl33d3r** [comment]
Forgot to reference this issue in commit https://github.com/byt3bl33d3r/CrackMapExec/commit/ab611c4d52d5228be4176bccc01e434d33259d4e, basically this module will be able to dump stored IE passwords among other credentials. Currently theres a bug in the Powersploit script tho. Waiting on a fix before merging to master

**Comment by unl1k3ly** [comment]
Thanks for the update.  Awesome work mate! Cheers

**Comment by byt3bl33d3r** [comment]
@AlessandroZ I'll probably be creating a module for LaZagne as well, think i was running into issues injecting the binary into memory using PowerShell cause the binary was compiled with DEP support.

**Comment by AlessandroZ** [comment]
Good news ! It could be useful to retrieve passwords from other applications, not only Chrome.

**Comment by byt3bl33d3r** [comment]
@unl1k3ly IE and Chrome enumeration has been added in v4.0 through the [mimikatz_enum_chrome](https://github.com/byt3bl33d3r/CrackMapExec/blob/master/cme/modules/mimikatz_enum_chrome.py), [mimikatz_enum_vault_creds](https://github.com/byt3bl33d3r/CrackMapExec/blob/master/cme/modules/mimikatz_enum_vault_creds.py), and the [enum_chrome](https://github.com/byt3bl33d3r/CrackMapExec/blob/master/cme/modules/enum_chrome.py) modules.

You can take a quick look at the source to understand what the diffrence is between the two chrome modules ;)

Let me know how it works for you!

**Comment by byt3bl33d3r** [comment]
@AlessandroZ I'm closing this and opening up another issue for LaZagne integration.
Also have you ever successfully injected LaZagne in memory through PowerShell?

**Comment by AlessandroZ** [comment]
No I have never tried to inject Lazagne using powershell. My way to use it is always through a pupy shell. I have embed lazagne as a module, so it's not written on the disk and it's executed only on memory. 

Using the ps1_oneliner payload, pupy can be downloaded and executed using powershell in memory without any file written on the disk. So then I use Lazagne. 

So on a classic security assessment, I use your tool or another method to deploy a pupy shell on the target and then I tried to post exploit. 

Let me know, if I can help you in anything, but I will understand as well that it could be complicated to embed Lazagne to your project.

**Comment by Greenwolf** [comment]
> @AlessandroZ I'm closing this and opening up another issue for LaZagne integration.
> Also have you ever successfully injected LaZagne in memory through PowerShell?

Hi @byt3bl33d3r, did you ever get around to implementing this? Having Lazagne is CME would be super useful! Also thanks for all your awesome tools 😄
